patches/server/0860-Furnace-RecipesUsed-API.patch

```diff
@@ -0,0 +1,43 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Jake Potrebic <jake.m.potrebic@gmail.com>
+Date: Thu, 13 Jan 2022 15:20:47 -0800
+Subject: [PATCH] Furnace RecipesUsed API
+
+
+diff --git a/src/main/java/org/bukkit/craftbukkit/block/CraftFurnace.java b/src/main/java/org/bukkit/craftbukkit/block/CraftFurnace.java
+index b6ee228d256e8dec0bcbd816a6dbaf53b47e5c8d..635f432c5ba6d91c199a43fb49a0e39dba622c28 100644
+--- a/src/main/java/org/bukkit/craftbukkit/block/CraftFurnace.java
++++ b/src/main/java/org/bukkit/craftbukkit/block/CraftFurnace.java
+@@ -73,5 +73,32 @@ public abstract class CraftFurnace<T extends AbstractFurnaceBlockEntity> extends
+         snapshot.cookSpeedMultiplier = multiplier;
+         snapshot.cookingTotalTime = AbstractFurnaceBlockEntity.getTotalCookTime(this.isPlaced() ? this.world.getHandle() : null, snapshot.recipeType, snapshot, snapshot.cookSpeedMultiplier); // Update the snapshot's current total cook time to scale with the newly set multiplier
+     }
++
++    @Override
++    public int getRecipeUsedCount(org.bukkit.NamespacedKey furnaceRecipe) {
++        return this.getSnapshot().recipesUsed.getInt(org.bukkit.craftbukkit.util.CraftNamespacedKey.toMinecraft(furnaceRecipe));
++    }
++
++    @Override
++    public boolean hasRecipeUsedCount(org.bukkit.NamespacedKey furnaceRecipe) {
++        return this.getSnapshot().recipesUsed.containsKey(org.bukkit.craftbukkit.util.CraftNamespacedKey.toMinecraft(furnaceRecipe));
++    }
++
++    @Override
++    public void setRecipeUsedCount(org.bukkit.inventory.CookingRecipe<?> furnaceRecipe, int count) {
++        final net.minecraft.resources.ResourceLocation location = org.bukkit.craftbukkit.util.CraftNamespacedKey.toMinecraft(furnaceRecipe.getKey());
++        java.util.Optional<? extends net.minecraft.world.item.crafting.Recipe<?>> nmsRecipe = (this.isPlaced() ? this.world.getHandle().getRecipeManager() : net.minecraft.server.MinecraftServer.getServer().getRecipeManager()).byKey(location);
++        com.google.common.base.Preconditions.checkArgument(nmsRecipe.isPresent() && nmsRecipe.get() instanceof net.minecraft.world.item.crafting.AbstractCookingRecipe, furnaceRecipe.getKey() + " is not recognized as a valid and registered furnace recipe");
++        if (count > 0) {
++            this.getSnapshot().recipesUsed.put(location, count);
++        } else {
++            this.getSnapshot().recipesUsed.removeInt(location);
++        }
++    }
++
++    @Override
++    public java.util.Collection<org.bukkit.NamespacedKey> getRecipesUsed() {
++        return this.getSnapshot().recipesUsed.keySet().stream().map(org.bukkit.craftbukkit.util.CraftNamespacedKey::fromMinecraft).toList();
```

Comment:
Shouldn’t this return a Set?

Reply:
it could be, but idk if it really matters. It's an unmodifiable list anyways.